### PR TITLE
bugfix - update wait_for_workflow to work with no workflow

### PR DESCRIPTION
**Pull request by barmagnezi**

### CODE REVIEW

File: src/commands/trigger-workflows-for-all-modified.yml [hunk 1]
```diff
@@ -269,6 +269,7 @@ steps:
           workflow_id=$(jq -r .id /tmp/swissknife/triggered_workflow.json)
           echo ""
           echo "Waiting for workflow: ${workflow_id} to complete"
+          sleep << parameters.sleep-time-between-checks >>
```
Comment (dany74q) [inline src/commands/trigger-workflows-for-all-modified.yml]: Is this b/c it takes time for a triggered workflow to return a proper response in the API below ? we're sleeping here before our first try, which is potentially unnecessary

File: src/commands/trigger-workflows-for-all-modified.yml [hunk 2]
```diff
@@ -289,7 +290,10 @@ steps:
             running=$(jq -s 'map(select(.status == "running")) | length' ${workflow_status})
             success=$(jq -s 'map(select(.status == "success")) | length' ${workflow_status})
             echo "total: ${total}, running: ${running}, success: ${success}"
-            if (( $total == 0 || $running > 0 )); then
+            if (( $total == 0 && $running == 0 )); then
```
Comment (dany74q) [inline src/commands/trigger-workflows-for-all-modified.yml]: running is a derivative of total, checking total == 0 should be enough

Comment (dany74q) [reply src/commands/trigger-workflows-for-all-modified.yml]: I'd also check that next_page_token == null

File: src/commands/trigger-workflows-for-all-modified.yml [hunk 3]
```diff
@@ -289,7 +290,10 @@ steps:
...
+              echo "No running workflows, return success"
+              exit 0
+            elif (( $total == 0 || $running > 0 )); then
```
Comment (dany74q) [inline src/commands/trigger-workflows-for-all-modified.yml]: Given the change above, is there still a good reason to sleep if total == 0 ?

**Comment by dany74q** [comment]
For reference:

```
# Existing workflow
curl -s https://circleci.com/api/v2/pipeline/e97b3858-2322-4c05-8d92-83fea28c9d55/workflow\?circle-token\=pasten | jq -r '.items[]?'
{
  "pipeline_id": "e97b3858-2322-4c05-8d92-83fea28c9d55",
  "id": "98ef58d8-d381-42fb-94af-016dc1452d35",
  "name": "trigger-needed-workflows",
  "project_slug": "gh/wiz-sec/customer-files",
  "status": "running",
  "started_by": "4dd6b66a-cbfc-4188-8da5-0b4ec220b0c2",
  "pipeline_number": 1549,
  "created_at": "2021-12-12T12:37:21Z",
  "stopped_at": null
}

# Non existent workflow
curl -s https://circleci.com/api/v2/pipeline/6596a10a-7404-4f4b-a4eb-59d78916a6d1/workflow\?circle-token\=pasten
{
  "next_page_token" : null,
  "items" : [ ]
}
```

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 2 above)
Comment (or-shachar) [reply src/commands/trigger-workflows-for-all-modified.yml]: +1

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 1 above)
Comment (or-shachar) [reply src/commands/trigger-workflows-for-all-modified.yml]: indeed... 
Maybe we can add another if to skip the initial sleep
```sh
if (( $total == 0 )); then
  sleep << parameters.sleep-time-between-checks >>
fi
```

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 3 above)
Comment (or-shachar) [reply src/commands/trigger-workflows-for-all-modified.yml]: probably needs to be `total ==0 && running >0`

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 2 above)
Comment (barmagnezi) [reply src/commands/trigger-workflows-for-all-modified.yml]: All pages read in "pagination loop" (line 278).
I think always next_page_token will be null.

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 3 above)
Comment (barmagnezi) [reply src/commands/trigger-workflows-for-all-modified.yml]: Fixed

File: src/commands/trigger-workflows-for-all-modified.yml (hunk 1 above)
Comment (barmagnezi) [reply src/commands/trigger-workflows-for-all-modified.yml]: Done

**Comment by dany74q** [comment]
Let's merge this @barmagnezi  ?

**Review by dany74q — APPROVED** [review]

**Pull request merged**